pandas/io/parsers/c_parser_wrapper.py

```diff
@@ -395,13 +413,15 @@ def _concatenate_chunks(chunks: list[dict[int, ArrayLike]]) -> dict:
     return result
 
 
-def ensure_dtype_objs(dtype):
+def ensure_dtype_objs(
+    dtype: DtypeArg | Mapping[Hashable, DtypeArg] | None
+) -> DtypeObj | Mapping[Hashable, DtypeObj] | None:
```

Comment:
`Mapping` -> `dict` for the return type. And in this case also for the input argument as the function specifically `isinstance`-checks for a `dict`.

Reply:
Good point, thx